dataobj/schedule.h

```diff
@@ -104,28 +109,34 @@ class schedule_t
 	/**
 	 * Get current stop of the schedule.
 	 */
-	uint8 get_current_stop() const { return current_stop; }
+	uint8 const& get_current_stop() const { return current_stop; }
 
 	/// returns the current stop, always a valid entry
 	schedule_entry_t const& get_current_entry() const { return current_stop >= entries.get_count() ? dummy_entry : entries[current_stop]; }
 	
-	schedule_entry_t const& get_next_entry() const;
+	schedule_entry_t const& get_next_entry();
 
 	/**
 	 * Set the current stop of the schedule .
 	 * If new value is bigger than stops available, the max stop will be used.
 	 */
-	void set_current_stop(uint8 new_current_stop) {
+	void set_current_stop(const uint8 new_current_stop) {
 		current_stop = new_current_stop;
 		make_current_stop_valid();
 	}
 
 	/// advance current_stop by one
-	void advance() {
-		if(  !entries.empty()  ) {
-			current_stop = (current_stop+1)%entries.get_count();
-		}
-	}
+	void advance();
+
+	// read next_line and return the entry of next_line
+	void advanced_entry(const uint8 advance_stop_number, uint8& result_entry_number, schedule_t* _schedule);
+	
+	// next_line setting
+	void set_next_line( linehandle_t l );
+	void unset_next_line();
+	uint16 get_next_line_id() {return next_line_id;}
+	// next_line condition is ok?
+	bool is_next_line();
```

Comment:
```suggestion
	bool next_line_exists() const;
```

`linehandle_t` のis_bound()チェックとwaypointチェックを行うようにしてください